p2p/protocol/identify/id.go

```diff
@@ -418,6 +415,11 @@ func (ids *IDService) consumeObservedAddress(observed []byte, c inet.Conn) {
 		return
 	}
 
+	if !HasConsistentTransport(maddr, ids.Host.Addrs()) {
+		log.Debugf("ignoring observed multiaddr that doesn't match the transports of any addresses we're announcing", c.RemoteMultiaddr())
+		return
+	}
```

Comment:
Do we care about the expansion of relay addrs we are advertising? Some of them maybe transient, because we happened to connect to some relay in order to talk to another NATed peer. But this is not a relay we are keeping connected because we selected it (and soon RESERVE capacity), but rather it's an incidental connection.

To be more specific here: If we are NATed/relayed and we connect to another NATed/relayed host, we will end up adding the other peer's relay into our observed address set and end up advertising that together with our primary relay connections.

Reply:
IMO, we should just get better at pushing updates. I actually like automatically discovering new relays like this as it helps spread the load a bit.

In the meantime, you're probably right. We don't want to advertise these addresses. Luckily, the default relay address filter _should_ filter out all those addresses at the next layer up.

Reply:
Yeah, I have big hopes with that patch.
The issue with using the random relays we happened to be connected is that these connections are not expected to be stable. 
It's great that we are spreading the load though.